File: proj3a/main.java
public class main {
    public static void main(String[] args){
        int vetor[] = new int[100];
        int n;
        func f = new func();

        for (int i = 1; i < vetor.length; i++) {
            vetor[i-1] = i;
        }

        n = f.permuta(vetor, 0, 8);
        System.out.println(n);
   }
}

File: proj3a/main.c
#include <stdio.h>

void troca(int vetor[], int i, int j){

	int aux = vetor[i];
	vetor[i] = vetor[j];
	vetor[j] = aux;
}

void permuta(int vetor[], int inf, int sup,int num){

	if(inf == sup){
		for(int i = 0; i <= sup; i++)
			printf("%d ", vetor[i]);
		printf("\n");
	}
	else{
		for(int i = inf; i <= sup; i++){
			troca(vetor, inf, i);
			permuta(vetor, inf + 1, sup, num++);
			troca(vetor, inf, i); 
		}
	}
}

int main(void)
{
	int v[100];

    for(int i = 1; i <= 100;i++){
        v[i-1] = i;
    }

	permuta(v, 0, 9, 1);

	return 0;
}